gemrb/core/System/VFS.cpp

```diff
@@ -110,11 +124,13 @@ static dirent* readdir(DIR *dir) {
 		}
 	}
 
-	auto& de = dir->entry;
-	auto n = wcstombs(const_cast<char*>(de.d_name), c_file.name, de.buffer.length() - 1);
-	de.buffer[n] = 0;
+	char16_t *c = reinterpret_cast<char16_t*>(c_file.name);
+	size_t n = 0;
+	for (; n < _MAX_PATH && *c != u'\0'; ++c, ++n) {}
```

Comment:
deems like there should be a std function for this, no?

Reply:
If there was a `std::find_n` ... maybe, I'm not aware of one r/n.

Reply:
I still haven't come up with anything std'ish that takes both conditions into account. Let me know if you find something but let's leave that for now.